eng/Analyzers.targets

```diff
@@ -14,7 +14,8 @@
   <PropertyGroup Condition="'$(RunAnalyzers)' != 'false'">
     <EnableSingleFileAnalyzer Condition="
       '$(EnableSingleFileAnalyzer)' == '' And
-      '$(TargetFrameworkIdentifier)' == '.NETCoreApp'">true</EnableSingleFileAnalyzer>
+      '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And
+      '$(IsGeneratorProject)' != 'true'">true</EnableSingleFileAnalyzer>
```

Comment:
What about IsReferenceSourceProject?

Reply:
Yes, I don't think we should have this analyzer enabled for test projects at the moment. So, I would prefer if you keep your change.